skyportal/models.py

```diff
@@ -80,14 +82,26 @@ class Stream(Base):
 User.groups = relationship('Group', secondary='group_users',
                            back_populates='users')
 
+
 @property
 def token_groups(self):
     return self.created_by.groups
+
+
 Token.groups = token_groups
 
 
 class Source(Base):
     id = sa.Column(sa.String, primary_key=True)
+    is_source = sa.Column(sa.Boolean, default=False)
+    is_candidate = sa.Column(sa.Boolean, default=True)
+    saved_as_source_by = relationship("User", back_populates="saved_sources")
+    saved_as_source_by_id = sa.Column(sa.ForeignKey("users.id"), nullable=True)
+    saved_as_source_at_time = sa.Column(ArrowType, nullable=True)
+    unsaved_as_source_at_time = sa.Column(ArrowType, nullable=True)
+    unsaved_as_source_by = relationship("User")
+    unsaved_as_source_by_id = sa.Column(sa.ForeignKey("users.id"), nullable=True)
```

Comment:
@acrellin @dannygoldstein we probably want a `candidate_marked_for_deletion` (default False) here too. That will help in the periodic vaccuuming up stale/bad candidates. Thoughts?